### PR TITLE
DEV: Add a `DVirtualList` windowing primitive to the ui-kit

### PR DESCRIPTION
**Pull request by megothss**
Adds `DVirtualList`, a windowing primitive for the ui-kit. It renders only the rows near the viewport, so a list of any length costs a viewport's worth of DOM. Chat is the first intended consumer, and a virtual table is the likely second.

A consumer supplies `@items` and `@estimateSize` and gets a scrolling list. The rest of the surface covers what a real list needs: stable keys, prepend anchoring, bottom-anchored live lists, edge callbacks for pagination, consumer-owned rows, opt-in ARIA roles, and an imperative handle via `@onRegisterApi`.

The engine is `@tanstack/virtual-core`, reached only through `ui-kit/lib/virtualizer.js`, so the dependency stays replaceable. Two styleguide examples cover fixed and measured row heights, and 27 test files cover windowing, anchoring, the edge latch, teardown, and consumer callbacks that throw.

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]

File: frontend/discourse/app/ui-kit/d-virtual-list.gts [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+                {{yield row.item row}}
+              {{else}}
+                {{! The role is dynamic, so the linter only sees a bare div and
+                    assumes the generic role. Both position attributes are gated at
+                    runtime on the roles that define them, and omitted otherwise. }}
+                {{! eslint-disable-next-line ember/template-no-unsupported-role-attributes }}
+                <div
+                  class="d-virtual-list__item"
+                  role={{@itemRole}}
+                  aria-setsize={{row.setSize}}
```
Comment (davidtaylorhq) [inline frontend/discourse/app/ui-kit/d-virtual-list.gts:698]: `-1` ok?

**Review by davidtaylorhq — APPROVED** [review]

File: frontend/discourse/app/ui-kit/d-virtual-list.gts (hunk 1 above)
Comment (megothss) [reply frontend/discourse/app/ui-kit/d-virtual-list.gts:698]: It's handled
